.github/actions/pull_images/action.yml

```diff
@@ -25,27 +25,32 @@ runs:
           df -h
         fi
 
-    - name: Pull images needed for default scenario
+    - name: Add images always needed
       shell: bash
       run: |
-        docker pull datadog/system-tests:proxy-v1 || true
-        docker pull postgres:latest
-    - name: Pull images needed for all scenarios
+        echo "services:" > compose.yaml
+        echo "  proxy:    {image: datadog/system-tests:proxy-v1}" >> compose.yaml
+        echo "  postgres: {image: postgres:latest}" >> compose.yaml
+
+    - name: Add images used in full list of scenario
       shell: bash
       run: |
         if [[ "${{ inputs.pull-all }}" == "true" ]]; then
-          docker pull mongo:latest
-          docker pull bitnami/kafka:3.1
-          docker pull bitnami/zookeeper:latest
-          docker pull cassandra:latest
-          docker pull rabbitmq:3-management-alpine
-          docker pull mysql/mysql-server:latest
-          docker pull mcr.microsoft.com/mssql/server:latest
-          docker pull mcr.microsoft.com/azure-sql-edge:latest
-          docker pull softwaremill/elasticmq:latest
-          docker pull datadog/system-tests:python_buddy-v0
-          docker pull datadog/system-tests:nodejs_buddy-v0
-          docker pull datadog/system-tests:java_buddy-v0
-          docker pull datadog/system-tests:ruby_buddy-v0
-          docker pull datadog/system-tests:golang_buddy-v0
+          echo "  mongo:          { image: "mongo:latest" }" >> compose.yaml
```

Comment:
At that point why not just write a real `docker-compose.yml` file and execute it instead of writing it programmatically? It would also make it easier to run the same compose file locally.

Reply:
Even better: dynamically create the compose file based on the scenario list you plan to execute on the machine.

* No need to maintain a file
* guarantee to load the exact needed list of images. 

* -> See my lasts commits